tools/extra/fpgadiag/nlb0.cpp

```diff
@@ -328,29 +328,39 @@ bool nlb0::setup()
     uint64_t nlb0_hi = swap64(uint64_t, n);
     uint64_t nlb0_lo = swap64(uint64_t, n+8);
 
-    do {
-        uint64_t feature_uuid_lo, feature_uuid_hi;
-        // Read the next feature header
-        dfh = accelerator_->read_csr64(static_cast<uint32_t>(offset));
-        feature_uuid_lo = accelerator_->read_csr64(static_cast<uint32_t>(offset+8));
-        feature_uuid_hi = accelerator_->read_csr64(static_cast<uint32_t>(offset+16));
-        if ((nlb0_lo == feature_uuid_lo)&&(nlb0_hi == feature_uuid_hi)) {
-            offset_ = offset;
-            printf("found the NLB offset=0x%x\n", offset);
-            break;
-        }
-        offset += NEXT_DFH_OFFSET(dfh);
-    } while (!DFH_EOL(dfh));
+    // enumerate dfh if input id arguement 
+    if (options_.find("id")) {
```

Comment:
This is a good catch, but the logic below is still flawed.  The issue is that, if there is no NLB0 AFU found, then we continue execution, and an assumption is made that offset_ (0) is valid.  I believe the fix is to add something like this:
```c++
bool found = false;
do {
...
if ((nlb0_lo == feature_uuid_lo) && (nlb0_hi == feature_uuid_hi)) {
   found = true;
   offset_ = offset;
   printf(...);
   break;
}
...
} while (!DFH_EOL(dfh));

if (!found) {
  <print error>
   return false;
}
```